### PR TITLE
Extend KerasWrapper to Functional API

### PR DESCRIPTION
**Pull request by huiwengoh**
Add functionality for KerasWrapper to support Keras functional API (in addition to sequential API).

Partially addresses #433

### CODE REVIEW

File: tests/test_frameworks.py [hunk 1]
```diff
@@ -153,13 +153,59 @@ def test_tensorflow(batch_size, shuffle_config, data=DATA, hidden_units=128):
...
+    preds_base = model.predict_proba(dataset_tf)
+
+    # Test CleanLearning performs well:
+    cl = CleanLearning(model)
+    cl.fit(dataset_tf, data["y"], clf_kwargs={"epochs": 10}, clf_final_kwargs={"epochs": 15})
+
+    preds = cl.predict(dataset_og_order)
+    err = np.sum(preds != data["y_og"]) / len(data["y_og"])
+    issue_indices = list(cl.label_issues_df[cl.label_issues_df["is_label_issue"]].index.values)
+    assert set(issue_indices).issubset(set(data["error_indices"]))
```
Comment (jwmueller) [inline tests/test_frameworks.py:198]: might this check start to fail if Keras releases new version with eg. different weight-initializations / optimizer (ie. how robust do you think it is)?  would it be more future proof if we instead check their intersection is nonempty or Jaccard similarity is at least `2 / len(data["error_indices"]` or something like that which doesn't require strict subset?

File: cleanlab/experimental/keras.py [hunk 2]
```diff
@@ -32,28 +32,86 @@
...
+from typing import Callable
 
 
-class KerasWrapper:
+class KerasWrapperModel:
     """
-    KerasWrapper is instantiated in the same way as a ``tf.keras.models.Sequential`` object,
+    KerasWrapperModel takes in a callable function to instantiate a Keras model (using Keras functional API)
+    that is compatible with :py:meth:`CleanLearning.fit()<cleanlab.classification.CleanLearning.fit>`.
+    """
```
Comment (jwmueller) [inline cleanlab/experimental/keras.py:42]:
```suggestion
    For using Keras sequential API, see `KerasWrapperSequential` class defined below.
    """
```

File: cleanlab/experimental/keras.py [hunk 3]
```diff
@@ -32,28 +32,86 @@
...
 import tensorflow as tf
 import numpy as np
+from typing import Callable
 
 
-class KerasWrapper:
+class KerasWrapperModel:
     """
-    KerasWrapper is instantiated in the same way as a ``tf.keras.models.Sequential`` object,
+    KerasWrapperModel takes in a callable function to instantiate a Keras model (using Keras functional API)
```
Comment (jwmueller) [inline cleanlab/experimental/keras.py:40]:
```suggestion
    KerasWrapperModel takes in a callable function to instantiate a Keras Model (using Keras functional API)
```

File: cleanlab/experimental/keras.py [hunk 4]
```diff
@@ -32,28 +32,86 @@
...
+    that is compatible with :py:meth:`CleanLearning.fit()<cleanlab.classification.CleanLearning.fit>`.
+    """
+
```
Comment (jwmueller) [inline cleanlab/experimental/keras.py:43]: Add a short docstring to describe some of the args below, especially the ones that are not standard Keras Model() init args. Also, a one-line real code example of what `model` might look like could be helpful

**Review by jwmueller — COMMENTED** [review]
Overall looks great! Just suggested a few minor changes

File: tests/test_frameworks.py (hunk 1 above)
Comment (huiwengoh) [reply tests/test_frameworks.py]: Updated to check for non-empty intersection

**Review by jwmueller — APPROVED** [review]
Great stuff!!

**Pull request merged**